test/metabase/query_processor/middleware/process_userland_query_test.clj

```diff
@@ -168,3 +172,31 @@
                 "val")))
         (testing "No QueryExecution should get saved when a query is canceled"
           (is (not @saved-query-execution?)))))))
+
+(deftest save-field-usage-test
+  (testing "execute a query will save field usages"
+    (qp.store/with-metadata-provider (mt/id)
+      (doseq [query-type [:mbql :mlv2]]
+        (let [;; randomize the binning so we can use it as the identifier for fieldusage down below
+              random-binning  (rand-int 13371337)
+              breakout-clause [:field (mt/id :products :price) {:binning {:num-bins  random-binning
+                                                                          :strategy  :num-bins
+                                                                          :bin-width random-binning}}]]
```

Comment:
If this query is really getting executed (which seems to be the case) 10M is a huge number of bins and bin width. `(rand-int 100)` will suffice here.

Reply:
I've updated the query so that it's not executed.
see https://github.com/metabase/metabase/pull/39671/commits/698fa2354051f611337751c345573bebc1d2a729